dev-python/pkgconfig/python_pkgconfig-1.5.5.recipe

```diff
@@ -4,58 +4,74 @@ Python 3.3+."
 HOMEPAGE="https://pypi.org/project/pkgconfig/"
 COPYRIGHT="201-2021 Mathias Vogelgesang"
 LICENSE="MIT"
-REVISION="3"
-SOURCE_URI="https://pypi.python.org/packages/source/p/pkgconfig/pkgconfig-$portVersion.tar.gz"
+REVISION="4"
+SOURCE_URI="https://files.pythonhosted.org/packages/source/p/pkgconfig/pkgconfig-$portVersion.tar.gz"
 CHECKSUM_SHA256="deb4163ef11f75b520d822d9505c1f462761b4309b1bb713d08689759ea8b899"
 SOURCE_DIR="pkgconfig-$portVersion"
 
-ARCHITECTURES="any"
+# Ideally, this should be "any", but we really want "pkg-config-x86" here, not the gcc2 version.
+# Python modules using this one have no concept of Haiku's x86_gcc2 vs x86, and we can pretty
+# much assume they only care for the non-gcc2 version of libs.
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
 	$portName = $portVersion
```

Comment:
I think, this needs to be changed to `python_pkgconfig$secondaryArchSuffix = $portVersion`. And maybe a "suffix-less" provide added conditionally for x86.

Reply:
I tend to agree with @jmairboeck here, this one is a bit of a weird one in the python recipes, so far only used by borgbackup. But it does search cross compiler (gcc2/gcc13) for library headers.
`$portName` on 32bit already refers to `python_pkgconfig_x86`, adding a suffix-less provides would make it handier to leave out the `$secondaryArchSuffix` on this?

Reply:
Personally I'd stay with "any", then check at runtime in python for the right executable at: https://github.com/matze/pkgconfig/blob/master/pkgconfig/pkgconfig.py#L108 
https://github.com/matze/pkgconfig/blob/master/pkgconfig/pkgconfig.py#L123

sys.platform.startswith("haiku") and platform.machine() == 'BePC'  then 'pkg-config-x86'

Reply:
IMO #10741 is correct, the build depends on this right tool.

Reply:
> We don't do that for other `ARCHITECTURES="all !x86_gcc2"` `dev-python/` recipes, AFAICS.
> 
> (see for example the recipes for `cffi`, `argon2-cffi-bindings`, `numpy`, `contourpy`).
> 
> So if that change is needed, then the same would apply for all the rest of recipes. I'd suggest opening an issue for that, so it is easier to keep track of progress.
> 
> FWIW, I haven't seen yet a "non _x86" `PROVIDES` conditionally added for the "main" package (only for `_python3x`). Do we ever add on "package A" a `REQUIRES` for "package_b" instead of a "package_b_python3x" version specific one? (the latter should all already provide the "non x86" names).

Ah, I missed the fact that this is just an empty base package and the real contents are in version-specific subpackages. The main package doesn't need a non-suffixed PROVIDE then. Sorry for the confusion.

Reply:
There may be a diffferent way to fix the problem here. It's possible to set the PKG_CONFIG_LIBDIR (if I remember the variable name correctly) to tell pkgconfig to search for .pc file in the lib/x86 directory. If you do this, you can then use the x86_gcc2 version of pkg-config to build software for the x86 architecture.

I'm not sure why we are not doing that generally (compiling pkg-config with x86 only, and using this variable to point it to the right place), but, if that's the only problem, maybe it is simpler to just set this environment variable in the recipes that use python_pkgconfig, and then the package itself will require no special case?

Reply:
> It's possible to set the PKG_CONFIG_LIBDIR (if I remember the variable name correctly

This should probably be: `PKG_CONFIG_PATH`, I use it sometimes in a cmake script to check out new things like:

`export PKG_CONFIG_PATH+=/boot/home/config/non-packaged/lib/pkgconfig/`

Reply:
Well it's just a python wrapper. Furthermore "pip install pkgconfig" won't install pkg-config for you.